pkg/datapath/linux/ipsec/ipsec_linux_test.go

```diff
@@ -82,6 +84,22 @@ func TestLoadKeys(t *testing.T) {
 	}
 }
 
+func TestLoadKeysIdentical(t *testing.T) {
+	log := setupIPSecSuitePrivileged(t)
+
+	keys := bytes.NewReader(keysDuplicateDat)
+	_, _, err := LoadIPSecKeys(log, keys)
+	require.NoError(t, err)
+}
+
+func TestLoadKeysSameSPI(t *testing.T) {
+	log := setupIPSecSuitePrivileged(t)
+
+	keys := bytes.NewReader(keysSameSpiDat)
+	_, _, err := LoadIPSecKeys(log, keys)
+	require.Error(t, err)
```

Comment:
It may be worth checking that we do fail with the expected error here. If the test starts failing with something else, it might hide a regression.